meta-facebook/grand-teton/src/platform/plat_hook.c

```diff
@@ -0,0 +1,211 @@
+	msg.target_addr = sensor_config[sensor_config_index_map[sensor_num]].target_addr;
+	msg.tx_len = 2;
+	msg.data[0] = 0x00;
+	msg.data[1] = pre_proc_args->vr_page;
+	if (i2c_master_write(&msg, retry)) {
+		printf("pre_isl69259_read, set page fail\n");
+		return false;
+	}
+	return true;
+}
+
+/* NVME pre read function
+ *
+ * set mux
+ *
+ * @param sensor_num sensor number
+ * @param args pointer to struct tca9548
+ * @param reading pointer to reading from previous step
+ * @retval true if setting mux is successful.
+ * @retval false if setting mux fails.
+ */
+bool pre_nvme_read(uint8_t sensor_num, void *args)
+{
+	if (!args)
+		return false;
+	if (!tca9548_select_chan(sensor_num, (struct tca9548 *)args))
+		return false;
+
+	return true;
+}
+
+/* AST ADC pre read function
+ *
+ * set gpio high if sensor is "SENSOR_NUM_VOL_BAT3V"
+ *
+ * @param sensor_num sensor number
+ * @param args pointer to NULL
+ * @param reading pointer to reading from previous step
+ * @retval true always.
+ * @retval false NULL
+ */
+bool pre_vol_bat3v_read(uint8_t sensor_num, void *args)
+{
+	ARG_UNUSED(args);
+
+	if (sensor_num == SENSOR_NUM_VOL_BAT3V) {
+		// gpio_set(A_P3V_BAT_SCALED_EN_R, GPIO_HIGH);
+		// k_msleep(1);
+	}
+
+	return true;
+}
+
+/* AST ADC post read function
+ *
+ * set gpio low if sensor is "SENSOR_NUM_VOL_BAT3V"
+ *
+ * @param sensor_num sensor number
+ * @param args pointer to NULL
+ * @param reading pointer to reading from previous step
+ * @retval true always.
+ * @retval false NULL
+ */
+bool post_vol_bat3v_read(uint8_t sensor_num, void *args, int *reading)
+{
+	ARG_UNUSED(args);
+	ARG_UNUSED(reading);
+
+	// if (sensor_num == SENSOR_NUM_VOL_BAT3V)
+	// 	gpio_set(A_P3V_BAT_SCALED_EN_R, GPIO_LOW);
+
+	return true;
+}
+
+/* INTEL PECI post read function
+ *
+ * modify certain sensor value after reading
+ *
+ * @param sensor_num sensor number
+ * @param args pointer to NULL
+ * @param reading pointer to reading from previous step
+ * @retval true if no error
+ * @retval false if reading get NULL
+ */
+
+bool post_cpu_margin_read(uint8_t sensor_num, void *args, int *reading)
+{
+	if (!reading)
+		return false;
+	ARG_UNUSED(args);
+
+	sensor_val *sval = (sensor_val *)reading;
+	sval->integer = -sval->integer; /* for BMC minus */
+	return true;
+}
+
+/* ADM1278 post read function
+ *
+ * modify ADM1278 power value after reading
+ *
+ * @param sensor_num sensor number
+ * @param args pointer to NULL
+ * @param reading pointer to reading from previous step
+ * @retval true if no error
+ * @retval false if reading get NULL
+ */
+bool post_adm1278_power_read(uint8_t sensor_num, void *args, int *reading)
+{
+	if (!reading)
+		return false;
+	ARG_UNUSED(args);
+
+	sensor_val *sval = (sensor_val *)reading;
+	float val = (float)sval->integer + (sval->fraction / 1000.0);
+
+	val = ADJUST_ADM1278_POWER(val);
+	sval->integer = (int)val & 0xFFFF;
+	sval->fraction = (val - sval->integer) * 1000;
+	return true;
+}
```

Comment:
This exact code is in Yv3.5-CL.
We should move it to common.

Reply:
We removed this function because it is used in yv35 but not used in this project, please review it again. Thanks.